src/internal/Transition.js

```diff
@@ -15,63 +13,84 @@ export const ENTERING = 2;
 export const ENTERED = 3;
 export const EXITING = 4;
 
-const propTypes = {
+type State = {
+  status: 0 | 1 | 2 | 3 | 4,
+}
+
+type DOMNode = Element | Text | null; // return type of ReactDOM.findDOMNode()
+
+type TransitionCallback = (node: DOMNode) => void;
+
+type DefaultProps = {
+  in: boolean,
+  unmountOnExit: boolean,
+  transitionAppear: boolean,
+  timeout: number,
+  onEnter: TransitionCallback,
+  onEntering: TransitionCallback,
+  onEntered: TransitionCallback,
+  onExit: TransitionCallback,
+  onExiting: TransitionCallback,
+  onExited: TransitionCallback,
+};
+
+type Props = DefaultProps & {
   /**
    * The content of the component.
    */
-  children: PropTypes.node,
+    children?: ReactElement<*>,
```

Comment:
This is an IDE formatting bug, fixed in the next minor release.  I'll correct this.